test/e2e/common_test.go

```diff
@@ -291,15 +294,43 @@ func PodmanTestCreateUtil(tempDir string, remote bool) *PodmanTestIntegration {
 	}
 	if remote {
 		p.PodmanTest.RemotePodmanBinary = podmanRemoteBinary
-		uuid := stringid.GenerateNonCryptoID()
+		var pathPrefix string
 		if !rootless.IsRootless() {
-			p.RemoteSocket = fmt.Sprintf("unix:/run/podman/podman-%s.sock", uuid)
+			pathPrefix = "/run/podman/podman"
 		} else {
 			runtimeDir := os.Getenv("XDG_RUNTIME_DIR")
-			socket := fmt.Sprintf("podman-%s.sock", uuid)
-			fqpath := filepath.Join(runtimeDir, socket)
-			p.RemoteSocket = fmt.Sprintf("unix:%s", fqpath)
+			pathPrefix = filepath.Join(runtimeDir, "podman")
 		}
+		// We want to avoid collisions in socket paths, but using the
+		// socket directly for a collision check doesn’t work; bind(2) on AF_UNIX
+		// creates the file, and we need to pass a unique path now before the bind(2)
+		// happens. So, use a podman-%s.sock-lock empty file as a marker.
+		tries := 0
+		for {
+			uuid := stringid.GenerateNonCryptoID()
+			lockPath := fmt.Sprintf("%s-%s.sock-lock", pathPrefix, uuid)
+			lockFile, err := os.OpenFile(lockPath, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0700)
+			if err == nil {
+				lockFile.Close()
+				p.RemoteSocketLock = lockPath
+				p.RemoteSocket = fmt.Sprintf("unix:%s-%s.sock", pathPrefix, uuid)
+				break
+			}
+			// For an unknown reason, we see fairly frequent collisions in the generated
+			// path names. So, collect quite a bit of information to hopefully help with that.
+			logrus.Warnf("RemoteSocket collision? on lock %#v, lock creation error %v", lockPath, err)
+			logrus.Warnf("GinkgoRandomSeed for PID %#v = %#v", os.Getpid(), GinkgoRandomSeed())
+			// stringid initializes the seed with something like this. It’s not likely to be very
+			// reproducible, stringid.init() probably ran a _long_ time ago, but let’s at least record the error,
+			// and maybe allow detecting unexpected low-entropy patterns.
+			cryptoseed, err := cryptorand.Int(cryptorand.Reader, big.NewInt(math.MaxInt64))
+			logrus.Warnf("cryptoseed (%v, %v), time %v", cryptoseed, err, time.Now().UnixNano())
+			tries++
+			if tries >= 1000 {
+				panic("Too many RemoteSocket collisions")
+			}
+		}
+		logrus.Infof("Chosen RemoteSocket %#v, PID %v", p.RemoteSocket, os.Getpid())
```

Comment:
IMHO this logging can be dropped.  We'll see the chosen socket immediately after in the test output.